lib/rubocop/cop/rspec/instance_variable.rb

```diff
@@ -67,6 +74,7 @@ def on_block(node)
 
           ivar_usage(node) do |ivar, name|
             return if inside_dynamic_class?(ivar)
+            return if inside_custom_matcher?(ivar)
```

Comment:
I'd just rename the `inside_dunamic_class?` method and modify the block inside to check for both cases. It would be easier to extend (no more conditions added here) and will perform only one lookup up in the tree.

example name:
`return if valid_usage?(ivar)`